firmware/controllers/tcu.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file        tcu.cpp
+ * @brief       Base classes for gear selection and transmission control
+ *
+ * @date Aug 31, 2020
+ * @author David Holdeman, (c) 2020
+ */
+
+#include "tcu.h"
+#include "tunerstudio_outputs.h"
+
+void TransmissionControllerBase::update(gear_e gear) {
+    setCurrentGear(gear);
+    postState();
+}
+
+gear_e TransmissionControllerBase::setCurrentGear(gear_e gear) {
+    currentGear = gear;
+    return getCurrentGear();
+}
+
+gear_e TransmissionControllerBase::getCurrentGear() {
+    return currentGear;
+}
+
+void TransmissionControllerBase::postState() {
+#if EFI_TUNER_STUDIO
+    tsOutputChannels.tcuCurrentGear = getCurrentGear();
+#endif
+}
+
+gear_e GearControllerBase::getDesiredGear() {
+    return desiredGear;
+}
+
+gear_e GearControllerBase::setDesiredGear(gear_e gear) {
+    desiredGear = gear;
+    return getDesiredGear();
```

Comment:
is this return value ever used?

Reply:
Not in this. I know I had a reason for making that the API, but I don't remember why. :)

Reply:
I think I wrote that before I decided which classes should get which functionality, and I thought maybe the class that inherited GearControllerBase might decide the request wasn't doable and return the current gear.

Reply:
Should I change this, or is there any value in leaving the flexibility?

Reply:
I would keep this time but not make flexible in the future

Reply:
yeah, probably doesn't hurt since it's not on a very hot code path, but also probably not necessary until we actually want to use it

Reply:
"not necessary until we actually want to use it" is the principle I try to live by